services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,89 @@
 package org.openkilda.functionaltests.spec.logging
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.openkilda.testing.service.elastic.model.KildaTags
+import org.openkilda.testing.service.elastic.model.KildaTypes
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+    String discoveryMessage = "push discovery package via"
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def result
+        if (this.profile.equalsIgnoreCase("virtual")) {
```

Comment:
1. Is this still required? I thought we have sorted out all inconsistencies on environment level
2. If this is still the case, then we should think towards implementing 2 versions of service (for virtual and for hardware profile) with the same interface but different implementations, instead of using if/else everywhere. But I more lean towards fixing this on logstash level

Reply:
There is a point in this, but as the logging is now fixed on local env, I'd just delete the separate branch